nifty-ssl/src/main/java/com/facebook/nifty/ssl/TicketSeedParser.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (C) 2012-2013 Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.nifty.ssl;
+
+import com.google.common.base.Throwables;
+import com.google.common.io.Files;
+import com.google.gson.Gson;
+import com.google.gson.JsonIOException;
+import com.google.gson.JsonSyntaxException;
+import com.google.gson.annotations.SerializedName;
+import org.apache.commons.codec.DecoderException;
+import org.apache.commons.codec.binary.Hex;
+import org.apache.tomcat.jni.SessionTicketKey;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.Reader;
+import java.nio.charset.Charset;
+
+/**
+ * To make distribution of ticket keys and rotating ticket keys easier, tickets can be distributed
+ * as a seed file in the following format
+ * {
+ *     "current": ["5afc6fb03ba4b15", "9547a3ab68b440ef7"],
+ *     "new": ["5afc6fb03ba4b15", "9547a3ab68b440ef7"],
+ *     "old": ["5afc6fb03ba4b15", "9547a3ab68b440ef7"]
+ * }
+ *
+ * The real ticket keys are generated from the seeds. The seeds can be arbitrary length hex encoded values.
+ * The current seeds are used to generate new tickets, and tickets encrypted with the old and new keys are
+ * accepted to allow for some leeway in rotation of tickets.
+ *
+ * The algorithm used to compute the session ticket encryption keys is the following:
+ *
+ * aesKey  = hkdf(seed, "aes")  -> truncated to AES bytes.
+ * hmacKey = hkdf(seed, "hmac") -> truncated to HMAC bytes
+ * name    = hkdf(seed, "name") -> truncated to name bytes
+ */
+public class TicketSeedParser {
+
+    public static byte[] defaultTicketSalt;
+    private static byte[] nameBytes = "name".getBytes();
```

Comment:
If you give an explicit Charset to use then I don't think it will behave differently in different locales (though I may be wrong). So: `nameBytes = "name".getBytes("UTF-8");`